node_modules/oae-core/resetpassword/js/resetpassword.js

```diff
@@ -0,0 +1,159 @@
+                // The username is still available
+                callback(false);
+            }
+        });
+    };
+    
+    /**
+     * Set up the validation on the register form, including the error messages
+     */
+    var setUpValidation = function() {
+        var validateOpts = {
+            'rules': {
+                'username': {
+                    'minlength': 3,
+                    'nospaces': true,
+                    'validchars': true,
+                    'usernameavailable': true
+                }
+            },
+            'messages': {
+                'username': {
+                    'required': oae.api.i18n.translate('__MSG__PLEASE_ENTER_YOUR_USERNAME__'),
+                    'minlength': oae.api.i18n.translate('__MSG__THE_USERNAME_SHOULD_BE_AT_LEAST_THREE_CHARACTERS_LONG__', 'register'),
+                    'nospaces': oae.api.i18n.translate('__MSG__THE_USERNAME_SHOULDNT_CONTAIN_SPACES__')
+                }
+            },
+            'methods': {
+                'validchars': {
+                    'method': function(value, element) {
+                        return this.optional(element) || !(/[<>\\\/{}\[\]!@#\$%\^&\*,:]+/i.test(value));
+                    },
+                    'text': oae.api.i18n.translate('__MSG__ACCOUNT_INVALIDCHAR__')
+                },
+                'usernameavailable': {
+                    'method': function(value, element) {
+                        var response = false;
+                        isUserNameAvailable(value, function(available) {
+                            response = available;
+                        });
+                        return response;
+                    },
+                    //'text': oae.api.i18n.translate('__MSG__THIS_USERNAME_HAS_ALREADY_BEEN_TAKEN__')
+                    'text': "Username does not exist."
+                }
+            },
+            'submitHandler': getSecret
+        };
+        oae.api.util.validation().validate($('#resetpassword-input-username', $rootel), validateOpts);
+    };
+		
+		/**
+		 * Get Secret: A start point for reseting passord
+		 */
+		var getSecret = function(form, event){
+
+          // Get the form values
+          var values = $('#resetpassword-input-username').serializeObject();
+          
+  				// disable redirect to other page
+  				event.preventDefault();
+
+  				var username = values.username;
+          
+  				$.ajax({
+  						'url': '/api/auth/local/reset/init/'+username,
+  						'type': 'GET',
+  						'data': null,
+  						'success': function(data) {
+  				
+  							//callback(null, data);
+  							oae.api.util.notification('Congratulations!','A Email has been sent to your Email address.');
+  						},
+  						'error': function(data) {
+                
+  							oae.api.util.notification('Sorry!','An error occured while sending the Email!');
+  						}
+  		});
+		};
+
+		/**
+		 * Show the resetpassword form
+		 */
+		var showResetForm = function() {
```

Comment:
Currently this modal will be shown behind the login modal - I assume this is because we've changed the login display from what it was when the PR was originally submitted, sorry.